[Kea Harvan Suyanto] iP

kea-S opened the pull request.

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

kea-S committed:
completed level 2 with prep for ToDo, Event, Deadline

kea-S committed:
bug fixing Interaction.start()

kea-S committed:
added the marking function

kea-S committed:
added functionality for differen types of classes

kea-S committed:
some exception handling implemented

kea-S committed:
delete functionality added

kea-S committed:
added saving

kea-S committed:
Merge branch 'branch-level-7'

Add save feature from branch-level-7

kea-S committed:
add minimal date and time functionality

kea-S committed:
Merge branch 'branch-level-8'

Add minimal date and time functionality from branch-level-8

kea-S committed:
Add storage class

kea-S committed:
Merge branch 'more-oop'

Add storage class

kea-S committed:
packaged shep

kea-S committed:
Merge branch 'add-gradle-support'

Add gradle support

kea-S committed:
Test gradle run

kea-S committed:
add j-unit-testcases

kea-S committed:
Merge branch 'branch-junit-tests'
add j-unit testcases

kea-S committed:
prod ready

kea-S committed:
Add some JavaDocs

kea-S committed:
Follow coding standard

kea-S committed:
Add find capability

kea-S committed:
Merge branch 'branch-A-CodingStandard'

Follow coding standard

kea-S committed:
Merge branch 'branch-A-JavaDoc'

Add JavaDoc

kea-S committed:
Fix conflicts

kea-S committed:
Add automated text UI testing

kea-S committed:
Merge branch 'branch-ATextUiTesting'

Add automated UI testing

kea-S committed:
Update README.md

kea-S committed:
Improved style

SomneelSaha2042 commented on src/main/java/shep/Shep.class:
You should add .class files to your gitignore, otw there is uneccessary bloat.

SomneelSaha2042 commented on src/main/java/shep/command/Commands.java:
As compared to 1 Commands class, we can try to extract the functionality out into multiple classes.

SomneelSaha2042 commented on src/main/java/shep/task/Deadline.java:
when parsing the date the code should be put in a try catch block to make sure we handle incorrect input

SomneelSaha2042 commented on src/main/java/shep/task/Event.java:
try catch block should be used

SomneelSaha2042 reviewed (commented):
LGTM! 
A few small issues here and there but otw seems g!

bkkuan commented on src/main/java/Interaction.java:
```diff
@@ -1,54 +1,28 @@
-import java.nio.file.Paths;
 
 public class Interaction {
     TaskList list;
```
Fields should generally be set to private for encapsulation

bkkuan commented on src/main/java/Interaction.java:
```diff
@@ -1,54 +1,28 @@
         this.list = new TaskList();
-
-        // load or initialise data directory
-        Path dataDirectoryPath = Paths.get("../data");
```
Its better to define constants for better maintainability
eg. private static final String DATA_DIRECTORY = "../data";
Then , Path dataDirectoryPath = Paths.get(DATA_DIRECTORY);

bkkuan commented on src/main/java/Interaction.java:
```diff
@@ -1,54 +1,28 @@
         boolean saidBye = false;
         while (!saidBye) {
             String currUserInputText = userInputScanner.nextLine();
-            saidBye = Commands.executeCommand(currUserInputText, this.list, this.filePath, true);
```
executeCommand() is called twice which may cause unintended behaviour.

bkkuan commented on src/main/java/TaskList.java:
```diff
@@ -1,10 +1,7 @@
 import java.io.IOException;
-import java.nio.file.Files;
-import java.nio.file.Path;
-import java.util.ArrayList;
```
java.util.ArrayList is imported twice. Remove the duplicate

bkkuan commented on src/main/java/TaskList.java:
```diff
@@ -42,53 +39,4 @@ public String toString() {
-
-    public void writeToFile(Path filePath) {
-        // overwrite previous taskList to be empty, no need to add anything to the file
-        try (FileWriter fileWriter = new FileWriter(filePath.toString(), false)) {
```
this block is redundant
FileWriter is opened but not used, which is pointless

bkkuan commented on src/main/java/TaskList.java:
```diff
@@ -1,10 +1,7 @@
-import java.nio.file.Path;
-import java.util.ArrayList;
 
 public class TaskList extends ArrayList<Task> {
```
generally a bad practice as it violates encapsulation

bkkuan reviewed (commented):
Code is well structured. Variables and method names follow Java conventions. Well done!

kea-S committed:
Add GUI functionality

kea-S committed:
Refactor code, remove Shep.java

Some code is not of quality. Mostly, SLAP. Shep.java is also not needed
anymore after moving over to GUI.

Refactor the code to improve code quality and remove Shep.java.

Code was refactored in order to adhere to SLAP. Several classes had
multiple layers of abstraction so method extraction had to be done.

kea-S committed:
Add assertations to reinforce logic

Some methods depended on the logic of other functions. To ensure that
the logic is followed  so that it can be utilised correctly by the class
Assertations were added.

Add Assertations to methods that require other methods to follow their
logic properly. Apart fromthe Commands class assertations never replaced
if else statements.

This is because assertations shouldn't be used to control the program
flow.

kea-S committed:
Fix bug in Commands

Logic bug regarding asserts replacing if-else statements caused program
to not be able to add tasks.

Reverted assert statements back to if-else.

If else statements worked before hand.

kea-S committed:
Merge pull request #2 from kea-S/branch-A-CodeQuality

Refactor code, remove Shep.java

kea-S committed:
Merge branch 'master' into branch-A-Assertions

Fix merge conflict

Several merge conflicts when trying to merge master with assertions
added.

Cleaned up.

Should be.

kea-S committed:
Merge pull request #3 from kea-S/branch-A-Assertions

Add assertations to reinforce logic

kea-S committed:
Add CI functionality

Add CI functionality using gradle.yml file under dir ./github/workflows

kea-S committed:
Add duplicate check on Task addition

On the addition of any type of Task into a TaskList, no duplicate checks
are made, allowing the chance for a duplicate task to be added into the
test list.

This is a redundant feature and duplicate Tasks shouldn't be allowed.

In order to fix this, a checkDuplicates was added in the TaskList
class as well as an equals method in Task.

Also updated TaskListTests and CommandsTest classes to account for newer
situations.

kea-S committed:
Merge checkDuplicates feature to main

On the addition of any type of Task into a TaskList, no duplicate checks
are made, allowing the chance for a duplicate task to be added into the
test list.

This is a redundant feature and duplicate Tasks shouldn't be allowed.

In order to fix this, a checkDuplicates was added in the TaskList
class as well as an equals method in Task.

Also updated TaskListTests and CommandsTest classes to account for newer
situations.

kea-S committed:
Add better exception handling

kea-S committed:
Add Better Exceptions

kea-S committed:
Merge to main, forgot to no-ff

kea-S committed:
Add more JavaDocs

kea-S committed:
Refactor Commands.java

kea-S committed:
Merge branch 'cut-executeCommand'
Refactor Commands.java

kea-S committed:
Fix several checkstyle issues

kea-S committed:
Fix several checkstyle issues

kea-S committed:
Add user guide

kea-S committed:
Merge branch 'user-guide'

Add User Guide

kea-S committed:
Fix user guide

kea-S committed:
Handle out of index error

kea-S committed:
Add contributor clarification

kea-S committed:
Ignore .class, .jar, .exe files
